web_src/js/features/repo-diff.ts

```diff
@@ -224,20 +227,46 @@ async function onLocationHashChange() {
   const attrAutoScrollRunning = 'data-auto-scroll-running';
   if (document.body.hasAttribute(attrAutoScrollRunning)) return;
 
-  const targetElementId = currentHash.substring(1);
-  while (currentHash === window.location.hash) {
-    // use getElementById to avoid querySelector throws an error when the hash is invalid
-    // eslint-disable-next-line unicorn/prefer-query-selector
-    const targetElement = document.getElementById(targetElementId);
-    if (targetElement) {
-      // need to change hash to re-trigger ":target" CSS selector, let's manually scroll to it
-      targetElement.scrollIntoView();
-      document.body.setAttribute(attrAutoScrollRunning, 'true');
-      window.location.hash = '';
-      window.location.hash = currentHash;
-      setTimeout(() => document.body.removeAttribute(attrAutoScrollRunning), 0);
+  // Check if this is a diff line selection hash (e.g., #diff-xxxL10 or #diff-xxxL10-R20)
+  const hashValue = currentHash.substring(1);
+  const range = parseDiffHashRange(hashValue);
+  if (range) {
+    // This is a line selection hash, try to highlight it first
+    const success = await highlightDiffSelectionFromHash();
```

Comment:
You should merge the old logic `targetElement.scrollIntoView();` into `highlightDiffSelectionFromHash`

Reply:
41857b4fcd3c9419d6e734137acca128c6406fa8